### PR TITLE
Make -view a flag to launch the report automatically after processing #315

### LINKED ISSUE #315: Make -view a flag to launch the report automatically after processing
-view is currently a separate command which user has to enter after 
the report is generated to view the report.

This is a hassle to the user as it can be joined together with the main
command to form an atomic action.

The original functionality of -view can remain though.

### PR DESCRIPTION
**Pull request by myteo**
Fixes #315 

Add -launch flag for users to view the report automatically after it has been generated

A separate -view command is required to view the report after generating it.

Lets
* Create an optional -launch flag for users to view the report automatically
* Keep the original functionality of -view

### CODE REVIEW

**Comment by yong24s** [comment]
LGTM.

Proposed commit message should observe a 72 characters line limit and it should consist of the problem, reason to change and step(s) taken to change it.

```
A separate -view command is required to view the report after
generating it. 

This is a hassle to the user as it can be joined together with the main
command to form an atomic action.

Let's add -launch flag for users to view the report automatically after
it has been generated.
```

File: src/main/java/reposense/parser/ArgsParser.java [hunk 1]
```diff
@@ -112,18 +118,21 @@ public static CliArguments parse(String[] args) throws ParseException {
             Optional<Date> untilDate = results.get("until");
             List<String> formats = results.get("formats");
             List<String> locations = results.get("repos");
+            Boolean isLaunchingAutomatically = results.get("launch");
```
[the remark this answers is not in the document]
Comment (myteo) [reply src/main/java/reposense/parser/ArgsParser.java]: When I am finished with this PR, I can create a separate issue and PR to address this.

File: src/main/java/reposense/model/CliArguments.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
     protected Optional<Date> sinceDate;
     protected Optional<Date> untilDate;
     protected List<String> formats;
+    protected Boolean isAutomaticallyLaunching;
```
[the remark this answers is not in the document]
Comment (myteo) [reply src/main/java/reposense/model/CliArguments.java]: Not really. I will change it to the primitive version.

**Comment by myteo** [comment]
Ok, I will change the implementation to reuse the existing `-view` flag instead. Since the implementation is completely different, I will do it in a separate PR and close this PR later.

**Comment by yamidark** [comment]
>Alternatively, we can also allow 0 arguments when view is used by itself and set a default

Yes, that may work for now in this PR.

In the future, we may want to separate the `-config/-repo/-view` into separate sub-commands which take their own parameters.

**Comment by myteo** [comment]
One issue I ran into:

Since `-config` argument has a default value which is the cwd, it is not easy to differentiate between 
- `-config -view` (`-config` used together with `-view`) 
- `-view` (used alone)

This is because in both cases, `-config` will have a value and not be `null`.

I can only implement `-view` flag for `-repos` and not `config`, but that seems undesirable.
Or I can also remove the default value for `-config`, which I don't think is a good idea either.

**Comment by eugenepeh** [comment]
`-config -view` hmm, you mean argparser does not handle this case and `-view` gets consume as the arg for `-config`?

**Comment by myteo** [comment]
> `-config -view` hmm, you mean argparser does not handle this case and `-view` gets consume as the arg for `-config`?

Nope. I probably wasn't clear. What I meant was there might be a problem with these 2 use cases:
1) If the `-view` parameter and `-config` parameter are both being used, eg. `java -jar RepoSense.jar -config ./my_configs -view`. [using -view as a flag to automatically launch after processing]
2) If the `-view` parameter is used alone. eg. `java -jar RepoSense.jar -view reposense-report` [launch report]

Because `-config` parameter has a default value which is the cwd, when we parse the command line arguments, in both use cases the values for `-config` and `view` will be non-null and it might not be easy to differentiate between the 2 use cases.

This problem occurs because we are adding use case 1, which is currently not a valid command.

**Comment by eugenepeh** [comment]
I see.. in this case, I think we can simply check whether arg exist in `-view`. 
If there is, 
-> show report and do nothing else
else 
-> perform the analysis and automatically open the report afterwards.

**Comment by myteo** [comment]
Ok, got it. How about the case where the user does something like:
`java -jar RepoSense.jar -config ./my_configs -view reposense-report`

I should perform analysis first right.

**Comment by eugenepeh** [comment]
> In this case, I just ignore -config argument and show report?

Yes. This would be the optimal solution. 
Can just explain in the user guide, something like -> `When an argument is supplied to the -view, it will not perform any analysis.`

**Comment by myteo** [comment]
New pull request: #446

**Pull request closed without merging**